pkg/pex/pex.go

```diff
@@ -358,6 +347,71 @@ func limit(opts *discovery.Options, is []peer.AddrInfo) []peer.AddrInfo {
 	return is
 }
 
+func (px *PeerExchange) namespace(nss string) namespace {
+	px.mu.Lock()
+	defer px.mu.Unlock()
+
+	ns, ok := px.ns[nss]
+
+	if !ok {
+		ns = px.createNamespace(nss)
+		px.ns[nss] = ns
+	}
+
+	return ns
+}
+
+func (px *PeerExchange) trackNamespace(nss string, ttl time.Duration) {
+	for {
+		select {
+		case px.namespace(nss).ttl <- ttl:
+			return
+		case <-px.namespace(nss).ctx.Done():
+		}
+	}
+}
+
+func (px *PeerExchange) createNamespace(nss string) namespace {
```

Comment:
Technically, this is a get-or-create operation.   Can we rename to `getOrCreateNamespace`?